### PR TITLE
Add finalfusion C++ library.

### PR DESCRIPTION
**Pull request by sebpuetz**
Also includes a test and an update to the most recent
finalfusion-ffi commit.

### CODE REVIEW

File: include/finalfusion_cxx.hh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#ifndef FINALFUSION_CXX_HH
```
Comment (danieldk) [inline include/finalfusion_cxx.hh]: Rename this file to `include/finalfusion-cxx/Embeddings.hh`

File: src/finalfusion_cxx.cc [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#include <cstdlib>
```
Comment (danieldk) [inline src/finalfusion_cxx.cc]: Rename file to `src/Embeddings.cc`.

File: src/finalfusion_cxx.cc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#include <memory>
+#include <stdexcept>
+
+#include "finalfusion.h"
+#include "finalfusion_cxx.hh"
```
Comment (danieldk) [inline src/finalfusion_cxx.cc]: `#include "finalfusion-cxx/Embeddings.hh`

**Comment by sebpuetz** [comment]
Pushed the changes, should we set up CI before merging this?

**Comment by danieldk** [comment]
On Mon, Sep 2, 2019, at 10:47, Sebastian Pütz wrote:
> Pushed the changes, should we set up CI before merging this?

I think it's also fine after this, then there is more to test ;).

**Pull request merged**